### PR TITLE
Fix #7818: Rating: Keyboard Support

### LINKED ISSUE #7818: Rating: Keyboard Support
**Describe the defect**
The rating component does not get the focus. So it can not be operated by keyboard and does not fulfill the WCAG 2.1 success criteria [2.1.1](https://www.w3.org/TR/WCAG21/#keyboard).

**Environment:**
 - PF Version: 10.0.0
 - Affected browsers: Chrome, Firefox, Edge

**To Reproduce**
1. Go to https://www.primefaces.org/showcase/ui/input/rating.xhtml?jfwid=55438
2. Navigate through the site using the tab key
3. Rating stars are skipped

**Expected behavior**
I would expect the rating stars to get the focus and that the selection can be made using the arrow keys.

### PR DESCRIPTION
**Pull request by jepsar**
Fix #7818

@weiprd please review

I did not add this to the what's new file as I'm not sure if this will end up in 11 or 12.

@mertsincan the focussed state does not look great. So this might require additional styling. Also, inline block might be more suitable for this component.

Screenshot 2021-10-02 at 09 12 13

### CODE REVIEW

File: primefaces/src/main/resources/META-INF/resources/primefaces/rating/rating.js [hunk 1]
```diff
@@ -147,10 +169,10 @@ PrimeFaces.widget.Rating = PrimeFaces.widget.BaseWidget.extend({
      */
     enable: function() {
         if(!this.isDisabled() || this.isReadOnly()) {
-           return; 
+            return;
         }
         this.cfg.disabled = false;
-
+        this.jq.attr('tabindex', this.tabindex);
```
Comment (melloware) [inline primefaces/src/main/resources/META-INF/resources/primefaces/rating/rating.js]: Why not add this line to the bindEvents and unbindEvents code?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by jepsar** [comment]
If this goes to 11, I need to add a note to 11's change log.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
This looks really good and is a great addition for accessibility.  @tandraschko I think this should go in 11.0.0 WDYT?

**Comment by melloware** [comment]
Can you rebase your PR i updated owasp-suppressions.xml to fix the error that is blocking your build.

**Comment by tandraschko** [comment]
im ok if mert does some styling for it
otherwise i feel its not complete

**Comment by jepsar** [comment]
@melloware I rebased, but now I no longer can run the showcase:

> java.lang.ClassNotFoundException: javax.faces.webapp.FacesServlet

> [WARNING] Failed startup of context o.e.j.m.p.JettyWebAppContext@5ff8e76a{/showcase,[file:///Users/jasper/GitHub/primefaces/primefaces-showcase/src/main/webapp/, jar:file:///Users/jasper/.m2/repository/org/primefaces/primefaces/11.0.0-SNAPSHOT/primefaces-11.0.0-SNAPSHOT.jar!/META-INF/resources],UNAVAILABLE}{file:///Users/jasper/GitHub/primefaces/primefaces-showcase/src/main/webapp/}
javax.servlet.UnavailableException: Class loading error for holder Faces Servlet==javax.faces.webapp.FacesServlet@87ee159b{jsp=null,order=-1,inst=false,async=false,src=DESCRIPTOR:file:///Users/jasper/GitHub/primefaces/primefaces-showcase/target/web.xml,STARTING}

**Comment by tandraschko** [comment]
profiles has been changed, check the readme

**Comment by jepsar** [comment]
OK, got it running now

**Comment by mertsincan** [comment]
@jepsar, Thanks a lot for your contribution!

**Pull request merged**